src/main/java/me/jellysquid/mods/lithium/common/chunk/SlimeChunkStorage.java

```diff
@@ -0,0 +1,37 @@
+package me.jellysquid.mods.lithium.common.chunk;
+
+import java.lang.Thread;
+import java.util.HashSet;
+
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.ChunkPos;
+ 
+public class SlimeChunkStorage{
+
+    private static HashSet<ChunkPos> slimeChunks = new HashSet<ChunkPos>();
+
+    public static boolean checkSlimeChunk(ChunkPos chunkChecking) {
+        return slimeChunks.contains(chunkChecking);
+    }
+
+    public static void addSlimeChunk(ChunkPos chunk) {
+        slimeChunks.add(chunk);
+    }
+
+    public static void unloadSlimeChunk(ChunkPos chunk) {
+        slimeChunks.remove(chunk);
+    }
+
+    public static void clearChunks(ServerWorld world) {
+        new Thread() {
```

Comment:
Does this need to be on another thread?

Reply:
Issue with this is that it is done each time a chunk is serialized because idk where else to put it so it will nuke performance iterating through each slime chunk in the world every time a chunk is saved.

Reply:
Yeah, I think the solution of storing it as a field on the chunk would work much better and avoid this problem altogether